test/chat/db/scope/cargo_key_scope_test.exs

```diff
@@ -1,41 +1,178 @@
 defmodule Chat.Db.Scope.CargoKeyScopeTest do
   use ExUnit.Case, async: false
 
-  alias Chat.Db.Scope.KeyScope
-  alias Chat.Dialogs
-  alias Chat.Identity
-  alias Chat.Messages
-  alias Chat.RoomInviteIndex
-  alias Chat.Rooms
-  alias Chat.User
-
-  test "room and invitations in dialogs should be selected" do
+  alias Chat.{
+    AdminRoom,
+    Db.Scope.KeyScope,
+    Dialogs,
+    Identity,
+    Messages,
+    RoomInviteIndex,
+    Rooms,
+    User,
+    Utils
+  }
+
+  @timeout 100
+
+  setup do
     {bot_key, room_key} = generate_user_with_dialogs_content_and_invite_in_room()
+    {:ok, %{bot_key: bot_key, room_key: room_key}}
+  end
+
+  test "room and invitations in dialogs should be selected", %{
+    bot_key: bot_key,
+    room_key: room_key
+  } do
+    assert_keys_for_cargo_keys(room_key, [bot_key], 1)
+  end
+
+  test "room invitation should be found in dialog of operator and user by only checkpoints key" do
+    {[checkpoints, checkpoint_keys], [operator], [user_c, user_d] = _users} =
+      setup_test_data(2, 1, 2)
+
+    assert Enum.count(checkpoints) == 2
+
+    context =
+      %{}
+      |> generate_cargo_room(operator)
+      |> put_dialogs_with_invites([checkpoints, operator], :index_1)
+      |> put_dialogs_with_invites([[user_c], operator], :index_2)
+      |> put_dialogs_with_invites([[user_d], user_c], :index_3)
+
+    room_key = context |> Map.get(:room_identity) |> Identity.pub_key()
+
+    assert_keys_for_cargo_keys(room_key, checkpoint_keys, 4)
+  end
+
+  test "room invitations found up to 1 handshake" do
+    {
+      [checkpoints, checkpoints_keys],
+      [operator_1, operator_2],
+      [user_1, user_2, user_3, user_4]
+    } = setup_test_data(3, 2, 4)
+
+    assert Enum.count(checkpoints) == 3
+
+    invitations_board =
+      %{}
+      |> generate_cargo_room(operator_1)
+      |> put_dialogs_with_invites([Enum.slice(checkpoints, 0..1), operator_1], :index_1)
+      |> put_dialogs_with_invites([[Enum.at(checkpoints, 2)], operator_2], :index_2)
+      |> put_dialogs_with_invites([[user_1], operator_1], :index_3)
+      |> put_dialogs_with_invites([[user_2, user_4], user_1], :index_4)
+      |> put_dialogs_with_invites([[user_3], user_2], :index_5)
+      |> put_dialogs_with_invites([[user_4, user_1], user_3], :index_6)
```

Comment:
`|> send_invite(from: [user_4, user_1], to: user_3, mark_as: :index_6)`

This looks more readable